АА - Add auth forms validation + error and success auth form state

AntonovAtanas opened the pull request:
- Add all auth forms validation
- Add error and success state for the auth forms
- Add Smolyan oblast + alphabetically sort the regions enum
- TODO: Auth services

![image](https://github.com/Getsov/Motorsport-Events-App/assets/114076833/5a271633-54ad-423d-8004-64969904ac9a)
![image](https://github.com/Getsov/Motorsport-Events-App/assets/114076833/494650a3-270a-4fd6-b38a-980e68aacaf4)

AntonovAtanas committed:
Edit Sofia enum property

AntonovAtanas committed:
Add error validation for login - email input

AntonovAtanas committed:
Add error validation for login password input

AntonovAtanas committed:
Add email, password, repeat password validation

AntonovAtanas committed:
add firstname and lastname validation

AntonovAtanas committed:
add organizator register email validation

AntonovAtanas committed:
Add organizer password and repeat password validation

AntonovAtanas committed:
Add red error clear icon

AntonovAtanas committed:
add organizator firstname and lastname validation

AntonovAtanas committed:
Merge branch 'dev' into AA-Login/Register-Pages

AntonovAtanas committed:
Add phone number input field with validation

AntonovAtanas committed:
Add success input icon

AntonovAtanas committed:
Add success state on form input

AntonovAtanas committed:
Add Smolyan to regions and sort the enum

AntonovAtanas committed:
Update auth-form-styles.scss

Getsov commented on src/app/user/login/login.page.ts:
```diff
@@ -9,4 +10,13 @@ export class LoginPage implements OnInit {
+    }
+
+    const { email, password } = loginFormData.value;
+    // TODO: userservice.register()
```
TODO -> .login()

Getsov commented on src/app/user/login/login.page.html:
```diff
@@ -1,16 +1,50 @@
 
   <div class="main-button-wrapper">
-    <button class="main-button">Вход</button>
+    <button class="main-button" [disabled]="loginForm.invalid" #button>Вход</button>
```
Why "#button" ? Using the element in the ts file or setting an ID ? Also check if it's like this, after clicking, if the page will reload? (It shouldn't)

AntonovAtanas replied:
#button is not needed. The form is not reloading the page

Getsov commented on src/app/user/organization-register/organization-register.page.html:
```diff
@@ -1,36 +1,142 @@
-      <ion-input type="password" fill="solid" label="Парола*" labelPlacement="floating" clearInput></ion-input>
+      <ion-input type="password" fill="solid" label="Парола*" labelPlacement="floating" clearInput name="password"
+        ngModel #password="ngModel" required minlength="8" clearInput
+        pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!\s).+$"></ion-input>
```
I think only min length and of course like 24 chars for max length is enough, without 1 small, 1 big letter, etc...

AntonovAtanas replied:
Fixed with the new patterns for password

Getsov commented on src/app/user/organization-register/organization-register.page.html:
```diff
@@ -1,36 +1,142 @@
+        Името трябва да съдържа само букви!
+      </p>
+      <p class="auth-validation-error-message" *ngIf="firstName.value && firstName.value.includes(' ')">
+        Не може да съдържа празни пространства!
```
Why not? I think there are names with two names inside?

AntonovAtanas replied:
Removed organizator first name and last name

Getsov commented on src/app/user/organization-register/organization-register.page.html:
```diff
@@ -1,36 +1,142 @@
+        Името трябва да съдържа само букви!
+      </p>
+      <p class="auth-validation-error-message" *ngIf="lastName.value && lastName.value.includes(' ')">
+        Не може да съдържа празни пространства!
```
Same here?

AntonovAtanas replied:
Removed first name and last name from organization form

Getsov commented on src/app/user/user-register/user-register.page.html:
```diff
@@ -5,30 +5,102 @@
-        <ion-input type="password" fill="solid" label="Парола*" labelPlacement="floating" clearInput></ion-input>
+        <ion-input type="password" fill="solid" label="Парола*" labelPlacement="floating" name="password" ngModel
+          #password="ngModel" required minlength="8" clearInput
+          pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!\s).+$"></ion-input>
```
Same here for the password validation

AntonovAtanas replied:
Fixed with the new patterns for password

Getsov commented on src/app/user/user-register/user-register.page.html:
```diff
@@ -5,30 +5,102 @@
+          Името трябва да съдържа само букви!
+        </p>
+        <p class="auth-validation-error-message" *ngIf="firstName.value && firstName.value.includes(' ')">
+          Не може да съдържа празни пространства!
```
Same here for firstName validation

AntonovAtanas replied:
It can have empty space and a dash now

Getsov commented on src/app/user/user-register/user-register.page.html:
```diff
@@ -5,30 +5,102 @@
       <div class="auth-input">
-        <ion-input type="text" fill="solid" label="Фамилия" labelPlacement="floating" clearInput></ion-input>
+        <ion-input type="text" fill="solid" label="Фамилия" labelPlacement="floating" name="lastName" clearInput ngModel
+          #lastName="ngModel" pattern="^[a-zA-Zа-яА-Я]+$"></ion-input>
```
Same here for lastName validation

AntonovAtanas replied:
It can have empty space and a dash now

Getsov commented on src/app/user/user-register/user-register.page.ts:
```diff
@@ -16,4 +17,13 @@ export class UserRegisterPage implements OnInit {
 
   ngOnInit() {}
+
+  onRegister(registerForm: NgForm) {
```
onRegisterSubmit

Getsov commented on src/app/user/login/login.page.ts:
```diff
@@ -9,4 +10,13 @@ export class LoginPage implements OnInit {
 
   ngOnInit() {}
+
+  onLogin(loginFormData: NgForm) {
```
onLoginSubmit

AntonovAtanas committed:
Add submit organization register form handler

AntonovAtanas committed:
Rename user register form handler

AntonovAtanas committed:
Rename login form handler

AntonovAtanas committed:
Fixed user register pass and names validation criteria

AntonovAtanas committed:
Update organizator register form

- Update password validation with the new criteria

- Organizator name can have any symbol. Only validation is for required

- Removed firstname and lastname for organizator

- Region moved to the bottom of the form

AntonovAtanas committed:
Removed not needed validation message for lastname

Getsov commented on src/app/user/organization-register/organization-register.page.html:
```diff
@@ -68,51 +58,27 @@
     <div class="auth-input">
       <ion-input type="text" fill="solid" label="Име на организатор*" labelPlacement="floating" clearInput
-        name="organizatorName" ngModel #organizatorName="ngModel" required pattern="^[a-zA-Zа-яА-Я0-9]+$"></ion-input>
+        name="organizatorName" ngModel #organizatorName="ngModel" required></ion-input>
```
Without common javascript special characters, like “<“, “/“, “>”

Getsov commented on src/app/user/organization-register/organization-register.page.ts:
```diff
@@ -14,4 +15,10 @@ export class OrganizationRegisterPage implements OnInit {
 
   ngOnInit() {}
+
+  onOrganizationRegisterSubmit(organizationRegister: NgForm) {
```
onOrganizerRegisterSubmit / onOrganizatorRegisterSubmit

Getsov commented on src/app/user/organization-register/organization-register.page.html:
```diff
@@ -1,4 +1,4 @@
-<form>
+<form #organizationRegister="ngForm" (submit)="onOrganizationRegisterSubmit(organizationRegister)">
```
organizer / organizator

AntonovAtanas replied:
Organizator seems more appropriate. Fixed

AntonovAtanas committed:
Avoid special characters in organization name

Getsov commented on src/app/user/user-register/user-register.page.html:
```diff
@@ -31,25 +31,15 @@
-        </p>
-        <p class="auth-validation-error-message" *ngIf="password.value && !password.value.match('[A-Z]')">
-          Паролата трябва да съдържа поне 1 голяма буква!
+          Паролата трябва да е с дължина поне 6 символа!
```
Do we have max length check?

AntonovAtanas replied:
Yes. Added error validation message for user registration and organizator registration

AntonovAtanas committed:
Add error validation message for maxlength password

AntonovAtanas committed:
Add error validation message and changed to organizator

- Add error validation message for maxlength password
- Edited to organizator form instead of organization

Getsov commented on src/app/user/organization-register/organization-register.page.ts:
```diff
@@ -16,8 +16,8 @@ export class OrganizationRegisterPage implements OnInit {
-  onOrganizationRegisterSubmit(organizationRegister: NgForm) {
-    if (organizationRegister.invalid) {
+  onOrganizatorRegisterSubmit(organizationRegisterForm: NgForm) {
+    if (organizationRegisterForm.invalid) {
```
organizator

AntonovAtanas committed:
Update form incoming data to organizatorRegisterForm
